frontend/main.js

```diff
@@ -57,12 +58,25 @@ async function startApp () {
   // this is to ensure compatibility between frontend and test/backend.test.js
   sbp('okTurtles.data/set', 'API_URL', window.location.origin)
 
-  // TODO: handle any socket errors!
-  createWebSocket(sbp('okTurtles.data/get', 'API_URL'), {
-    // TODO: verify these are good defaults
-    timeout: 3000,
-    strategy: ['disconnect', 'online', 'timeout']
-  })
+  // URL used to connect to the pubsub server. May include query parameters.
+  let pubsubURL = sbp('okTurtles.data/get', 'API_URL')
+
+  if (process.env.NODE_ENV === 'development') {
+    // This is temporarily used in development mode to help the server improve
+    // its console output until we have a better solution. Do not use for auth.
+    const debugID = Array.from(
+      // $FlowIssue doesn't support crypto yet: https://github.com/facebook/flow/issues/5019
+      crypto.getRandomValues(new Uint8Array(3)),
+      (byte) => byte.toString(16).padStart(2, '0')
+    ).join('')
+    pubsubURL += `?debugID=${debugID}`
+  }
+  sbp('okTurtles.data/set', PUBSUB_INSTANCE, createGIPubSubClient(
+    pubsubURL, {
+      // This option can be enabled since we are not doing auth via web sockets.
+      reconnectOnTimeout: true
```

Comment:
Here we lost the `timeout` value of 3000, could that be added back in please (especially since the default appears to be 5000)?

Reply:
That was on purpose, 3000 is likely too short:
- We have seen that timing issues can cause the Travis build to fail.
- All web socket libraries that I have checked use a higher default timeout or do not seem to specify any at all:
  - [reconnecting-websocket](https://github.com/pladaria/reconnecting-websocket#default-values): 4000
  - [Primus](https://github.com/primus/primus/#connecting-from-the-browser): 10000
  - [socket.io](https://socket.io/docs/v4/client-initialization/#timeout): 20000

Reply:
ok, it's still useful to specify it here though to remind us that's an option and what its current value is without diving into the implementation. To your first point, we can also extend it for travis:

```js
```

Reply:
ok, it's still useful to specify it here though to remind us that's an option and what its current value is without diving into the implementation. To your first point, we can also extend it for travis, e.g. something like this:

```js
timeout: process.env.NODE_ENV === 'development' ? 10000 : 5000
```